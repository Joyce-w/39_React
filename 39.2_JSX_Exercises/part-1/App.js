const App = () => {
    return (
       <div>
            <FirstComponent />
            <NamedComponent name="Joyce"/>
    </div>
   ) 
}

ReactDOM.render(<App/>, document.getElementById("root"))