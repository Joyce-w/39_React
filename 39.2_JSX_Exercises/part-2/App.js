const App = () => {
    return (
        <div>
             <Tweet username="canadianGirl" name="Mia" date="Jan 20, 2020" message="New video up!"/>
             <Tweet username="Colt" name="Steele" date="Jan 21, 2020" message="Free Tuition!"/>
             <Tweet username="joyjoy" name="joy" date="Mar 20, 2020" message="Nothing in my head"/>
        </div>
    )
   
}


ReactDOM.render(<App/>, document.getElementById("root"))