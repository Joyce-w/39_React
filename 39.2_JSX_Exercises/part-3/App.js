const App = () => {
    return (
        <div>
            <Person name="candy" age="19" hobbies={['eating', 'dining', 'playing']} />
                <Button text="First Button"/>
            <Person name="Alexandar" age="19" hobbies={['eating', 'dining', 'playing']} />
                <Button text="Click Me"/>
            <Person name="JungyoonLee" age="12" hobbies={['eating', 'dining', 'playing']} />
                <Button text="Last Button"/>
        </div>
    )
}


ReactDOM.render(<App/>, document.getElementById("root"))