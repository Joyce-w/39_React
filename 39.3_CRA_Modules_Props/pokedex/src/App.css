#App-header{
  text-align: center;
  color: rgb(172, 181, 189);
}