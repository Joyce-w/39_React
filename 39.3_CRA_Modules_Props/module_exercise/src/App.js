import logo from './logo.svg';
import './App.css';
import foods from './foods';
import { choice, remove } from './helpers'
function App() {
  let randFruit = choice(foods)
  console.log(`I'd like one ${randFruit} please`)
  console.log(`Here you go: ${randFruit} `)
  console.log(`Delicious, may I have another?`)
  let newArray = remove(foods, randFruit)
  console.log(`We're all out. We have ${newArray.length} left`)
  

  return (
    <div className="App">
      <header className="App-header">
        <img src={logo} className="App-logo" alt="logo" />
        <p>
          Edit <code>src/App.js</code> and save to reload.
        </p>
        <a
          className="App-link"
          href="https://reactjs.org"
          target="_blank"
          rel="noopener noreferrer"
        >
          Learn React
        </a>
      </header>
    </div>
  );
}

export default App;
