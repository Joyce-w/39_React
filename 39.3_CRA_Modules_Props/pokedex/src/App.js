
import './App.css';
import Pokedex from './Pokedex'
import pokemon from './pokemon'

function App() {
  return (
    <div className="App">
      <header className="App-header">
        <h1 id="App-header">Pokedex</h1>
        <div id="Pokedex">
          <Pokedex key={ pokemon.id} pokemon={ pokemon }/>
        </div>
        
      </header>
    </div>
  );
}

export default App;
