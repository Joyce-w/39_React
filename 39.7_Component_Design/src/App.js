import React from "react";
import Board from "./Board";
import "./App.css";

/** Simple app that just shows the LightsOut game. */

function App() {
  return (
    <div className="App">
      <table><Board nrows={4} ncols={4} chanceLightStartsOn={0.4}/></table>
    </div>
  );
}

export default App;
