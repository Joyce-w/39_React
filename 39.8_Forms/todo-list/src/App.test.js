import { render, screen } from '@testing-library/react';
import App from './App';

//smoketest
test('renders learn react link', () => {
  render(<App />);
});


//snapshot
it("matches snapshot", () => {
  const { asFragment } = render(<App />)
  expect(asFragment()).toMatchSnapshot();
})