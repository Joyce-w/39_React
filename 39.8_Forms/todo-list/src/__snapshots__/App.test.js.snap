// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <form
      class="NewTodoForm-form"
    >
      <label
        for="task"
      >
        New Todo 
      </label>
      <input
        class="NewTodoForm-input"
        id="task"
        name="task"
        placeholder="More things to do"
        type="text"
        value=""
      />
      <button
        class="NewTodoForm-btn"
      >
        Add new task
      </button>
    </form>
    <h3>
      What to do List
    </h3>
    <div
      class="NewTodoForm-ul-div"
    >
      <ul
        class="NewTodoForm-ul"
      >
        <div
          class="Todo"
          data-testid="Todo-test"
          id="1778c1a1-7cf1-4b8a-99e6-344ba18603d3"
        >
          <li
            class="Todo-li"
          >
            Clean fish tank
            <button
              class="Todo-btn"
            >
              x
            </button>
          </li>
        </div>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
